woocommerce/woocommerce-gateway-stripe
# Enable payment methods when enabling/disabling UPE

Fixes #1931

### Description
When enabling UPE, the enabled/disabled status of the UPE payment methods should reflect the previously enabled/disabled legacy methods.
When disabling UPE, the enabled/disabled status of the UPE payment methods should reflect the previously enabled/disabled legacy methods.

Giropay and iDeal is selected in the main settings page when UPE is disabled
![main settings page](https://user-images.githubusercontent.com/33387139/134866558-9e5e6939-0711-47b9-99ce-505c3d2c0615.png)

After enabling UPE from onboarding wizard, Giropay and iDeal checkboxes are checked 
![onboarding wizard checkboxes](https://user-images.githubusercontent.com/33387139/134866629-3c34eeff-de90-4969-ac9e-28ef51fe03ae.png)


### Proposed changes 

### Testing instructions
- Enable `_wcstripe_feature_upe_settings`

Test enabling
- Enable some legacy payment methods and save here `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout`
- Enable UPE from `http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard`
- Check if the corresponding UPE methods (previously selected in the legacy methods) are already selected

Test disabling
- Enable some more UPE methods and save from `http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard` or `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe`
- Disable UPE from `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe`
- Check if the corresponding legacy methods (previously selected UPE methods in new settings) are already selected here `http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout`
